cmd/zc_traverser_blob_account.go

```diff
@@ -116,21 +107,12 @@ func (t *blobAccountTraverser) Traverse(preprocessor objectMorpher, processor ob
 	return nil
 }
 
-func newBlobAccountTraverser(rawURL *url.URL, p pipeline.Pipeline, ctx context.Context, includeDirectoryStubs bool, incrementEnumerationCounter enumerationCounterFunc, s2sPreserveSourceTags bool, cpkOptions common.CpkOptions, preservePermissions common.PreservePermissionsOption) (t *blobAccountTraverser) {
-	bURLParts := azblob.NewBlobURLParts(*rawURL)
-	cPattern := bURLParts.ContainerName
-
-	// Strip the container name away and treat it as a pattern
-	if bURLParts.ContainerName != "" {
-		bURLParts.ContainerName = ""
-	}
-
+func newBlobAccountTraverser(serviceClient *service.Client, container string, ctx context.Context, includeDirectoryStubs bool, incrementEnumerationCounter enumerationCounterFunc, s2sPreserveSourceTags bool, cpkOptions common.CpkOptions, preservePermissions common.PreservePermissionsOption) (t *blobAccountTraverser) {
```

Comment:
Some of these options aren't always required. Should we take the initiative now to options bag this as best we can?

Reply:
this is a good idea, but I'd still prefer it to be it's own PR (maybe after blob traverser and cmd code are merged)

only reason is changing this results in a lot of test changes so it can get confusing to review.